modAionImpl/src/org/aion/zero/impl/sync/TaskImportBlocks.java

```diff
@@ -229,6 +229,8 @@ private SyncMode processBatch(SyncMode syncMode, List<Block> batch, String displ
                 syncStats.updatePeerBlocks(displayId, count, BlockType.IMPORTED);
                 log.info("<import-status: node = {}, from = #{}, to = #{}, time elapsed = {} ms>", displayId, first, currentBest, importDuration);
             }
+
+            chain.checkKernelShutdownForCLI();
```

Comment:
Instead of exposing this method and calling it here could the issue be fixed by making `enableFullSyncCheck` and `reachedFullSync` atomic boolean variables?